packages/vscode-extension/src/plugins/render-outlines/render-outlines-plugin.ts

```diff
@@ -22,80 +23,99 @@ const INSPECTOR_AVAILABILITY_MESSAGES = {
 export class RenderOutlinesPlugin implements ToolPlugin, RenderOutlinesInterface, Disposable {
   private eventEmitter = new EventEmitter();
   private isEnabled = false;
-  private devtoolsListeners: Disposable[] = [];
+  private eventListeners: Disposable[] = [];
   private inspectorAvailability: InspectorAvailabilityStatus =
     InspectorAvailabilityStatus.Available;
-  private wasPreviouslyEnabled = false;
+  private experimentalEnable: boolean;
+  private enableRequested = false;
+
+  public readonly id = RENDER_OUTLINES_PLUGIN_ID;
+  public readonly label = "Outline Renders";
+  public readonly toolInstalled = true;
+  public readonly persist = false;
 
   constructor(
     private inspectorBridge: RadonInspectorBridge,
-    private onAvailabilityChange: () => void
+    private onAvailabilityChange: () => void,
+    private workspaceConfigState: StateManager<WorkspaceConfiguration>
   ) {
-    this.devtoolsListeners.push(
+    this.experimentalEnable = workspaceConfigState.getState().enableExperimentalElementInspector;
+    this.setupEventListeners();
+  }
+
+  /**
+   * Sets up all event listeners for the plugin
+   */
+  private setupEventListeners(): void {
+    const disposables = [
       this.inspectorBridge.onEvent("appReady", () => {
         this.setEnabled(this.isEnabled);
-      })
-    );
-    this.devtoolsListeners.push(
+      }),
+
       this.inspectorBridge.onEvent("pluginMessage", ({ pluginId, type, data }) => {
         if (pluginId === RENDER_OUTLINES_PLUGIN_ID && type === "rendersReported") {
           this.eventEmitter.emit("rendersReported", data);
         }
-      })
-    );
-    this.devtoolsListeners.push(
+      }),
+
       this.inspectorBridge.onEvent(
         "inspectorAvailabilityChanged",
         (inspectorAvailability: InspectorAvailabilityStatus) => {
           this.inspectorAvailability = inspectorAvailability;
-
-          if (inspectorAvailability === InspectorAvailabilityStatus.Available) {
-            this.activateDueToAvailabilityChange();
-          } else {
-            this.deactivateDueToAvailabilityChange();
-          }
-
+          this.updatePluginState();
           this.onAvailabilityChange();
         }
-      )
-    );
+      ),
+
+      this.workspaceConfigState.onSetState((state) => {
+        this.experimentalEnable = !!state.enableExperimentalElementInspector;
+        this.updatePluginState();
+        this.onAvailabilityChange();
+      }),
+    ];
+
+    disposables.forEach((disposable) => {
+      this.eventListeners.push(disposable);
+    });
```

Comment:
NIT: `Array.push` can accept many arguments, you can shorten this to

```suggestion
    this.eventListeners.push(...disposables)
```
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/push#merging_two_arrays

Reply:
Love it, changed in 9ef984f6